### PR TITLE
Auto install lightbox without adding a <amp-lightbox-viewer> tag

### PR DESCRIPTION
**Pull request by cathyxz**
In other words, simplify the the lightbox API so that we install a lightbox viewer tag automatically when the extension loads.

### CODE REVIEW

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 1]
```diff
@@ -903,14 +904,39 @@ export class AmpLightboxViewer extends AMP.BaseElement {
...
+/**
+ * Tries to find an existing amp-lightbox-viewer, if there is none, it adds a
+ * default one.
+ * @param {!Window} win
+ * @return {!Promise}
+ */
+function installLightboxViewer(win) {
+  const ampdoc = Services.ampdocServiceFor(win).getAmpDoc();
+  // TODO (#12859): make this work for more than singleDoc mode
+  return waitForBodyPromise(/**@type {!Document}*/ (ampdoc.getRootNode()))
```
[the remark this answers is not in the document]
Comment (cvializ) [reply extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: yep! `ampdoc.whenBodyAvailable().then(body => {})`

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 2]
```diff
@@ -903,14 +904,39 @@ export class AmpLightboxViewer extends AMP.BaseElement {
  */
 export function installLightboxManager(win) {
   if (isExperimentOn(win, TAG)) {
-    // TODO(aghassemi): This only works for singleDoc mode. We will move
+    // TODO (#12859): This only works for singleDoc mode. We will move
```
Comment (cvializ) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: nice!

**Comment by cathyxz** [comment]
@cvializ nope, you shouldn't ever need to add a `<amp-lightbox-viewer>` element at all because the extension adds it automatically. All you need to do is include the extension and add the `lightbox` attribute to your images / carousels / anything you intend to lightbox.

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (cathyxz) [reply extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: So I remember looking into this before, but the `ampdoc-impl.js` files has this function as not implemented: 

```
  /**
   * Returns a promise that will be resolved when the ampdoc's body is
   * available.
   * @return {!Promise<!Element>}
   */
  whenBodyAvailable() {
    return /** @type {?} */ (dev().assert(null, 'not implemented'));
  }
```

which is why I didn't use it in the first place. Is this actually implemented somewhere else?

[the remark this answers is not in the document]
Comment (cathyxz) [reply extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: Oh nvm. I see that it's implemented in an override by `AmpDocSingle` and `AmpDocShadow` differently.

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 3]
```diff
@@ -903,14 +904,38 @@ export class AmpLightboxViewer extends AMP.BaseElement {
...
+ * @param {!Window} win
+ * @return {!Promise}
+ */
+function installLightboxViewer(win) {
+  const ampdoc = Services.ampdocServiceFor(win).getAmpDoc();
+  // TODO (#12859): make this work for more than singleDoc mode
+  return ampdoc.whenBodyAvailable().then(body => {
+    const existingViewer = elementByTag(ampdoc.getRootNode(), TAG);
+    if (!existingViewer) {
+      const matches = ampdoc.getRootNode().querySelectorAll('[lightbox]');
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: I would install it regardless. elements with `[lightbox]` maybe dynamically loaded through AMP list.

**Pull request merged**